src/stan/model/model_base.hpp

```diff
@@ -0,0 +1,612 @@
+#ifndef STAN_MODEL_MODEL_BASE_HPP
+#define STAN_MODEL_MODEL_BASE_HPP
+
+#include <stan/io/var_context.hpp>
+#include <stan/math/rev/core.hpp>
+#include <stan/model/prob_grad.hpp>
+#include <boost/random/additive_combine.hpp>
+#include <ostream>
+#include <string>
+#include <utility>
+#include <vector>
+
+namespace stan {
+namespace model {
+
+/**
+ * The base class for models defining all virtual methods required for
+ * services.  Any class extending this class and defining all of its
+ * virtual methods can be used with any of the Stan services for
+ * sampling, optimization, or variational inference.
+ *
+ * <p><i>Implementation Details:</i> The reason there are so many
+ * overloads of the `log_prob` and `write_array` methods is that
+ * template methods cannot be declared virtual.  This class extends
+ * `stan::model::prob_grad` in order to define sizing for the number
+ * of unconstrained parameters;  thus it is not a pure virtual base
+ * class.
+ *
+ *<p>The approach to defining models used by the Stan language code
+ * generator is use the curiously recursive template base class defined
+ * in the extension `stan::model::model_base_crtp`; see that class's
```

Comment:
minor: you can drop the text after the semicolon. Here's what the rendered doxygen looks like:
![image](https://user-images.githubusercontent.com/425751/61265105-fcb41180-a75c-11e9-9814-ee9fbd20079e.png)

(The `stan::model::model_base_crtp` link takes you to the right place.)

Reply:
Done.